### PR TITLE
setup.py: Delegate Python version check to coala

### PR DESCRIPTION
**Pull request by jayvdb**
Depends on https://github.com/coala/coala/pull/3384

### CODE REVIEW

**Comment by jayvdb** [comment]
crap. also depends on https://gitlab.com/coala/package_manager/issues/2 , sort of.

**Comment by jayvdb** [comment]
(Circle is failing because it is a Trusty build; I forgot to switch.)
Commit message and other stuff is needing improvement.
I would like a review of the concept before tidying it up.

File: .travis.yml [hunk 1]
```diff
@@ -74,25 +79,43 @@ before_install:
...
   - echo 'language_check==0.8.*' >> requirements.txt
+  # On unsupported versions, coala will be installed in the `script` block
+  - >
+    if [[ $TRAVIS_PYTHON_VERSION_MINOR < 4 || $TRAVIS_PYTHON_VERSION_MAJOR == 2 ]]; then
+      sed -e '/^coala/d' requirements.txt > requirements.new
+      mv requirements.new requirements.txt
+    fi
 
 before_script:
+  - mv requirements.orig requirements.txt
```
Comment (Mixih) [inline .travis.yml]: isn't this going to overwrite the custom requirements.txt made in line 90?

Comment (jayvdb) [inline .travis.yml]: setup.py reads `requirements.txt` , and needs the original, without all of the fiddles done above.

**Comment by sils** [comment]
@jayvdb I wonder if we need an own test build for that, is there any danger of regressions here? This should happen by default if setuptools doesn't mess up, right? So that's upstream functionality that we don't necessarily want to waste a lot of build time on.

**Comment by Mixih** [comment]
@sils can't we rely on the regression testing built into coala now?

**Comment by sils** [comment]
yep coala has tests for it so we don't need them in coala-bears

**Comment by Mixih** [comment]
ack dee94fd

**Comment by Mixih** [comment]
Ready to go after a rebase....

**Comment by jayvdb** [comment]
heavily revised.

**Comment by Mixih** [comment]
ack a0f258e

**Comment by jayvdb** [comment]
@rultor merge

**Pull request merged**